### PR TITLE
Patron2020 88/network library

### PR DESCRIPTION
**Pull request by Deelay1403**
I created a simple network function with this functionality:
- Establish a network connection.
- Configure static or dynamic IP address
- Change device MAC address

### CODE REVIEW

File: Network/Network.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#ifndef NETWORK_H
+#define NETWORK_H
+
+/*  TODO: change it to ssid and pass from config file*/
+#define STASSID "FIND ME"
+#define STAPASS "87654321"
```
Comment (kuba8686) [inline Network/Network.h:6]: Have to be coordinated with common config file. I will do it after approving this PR when doing common project structure merging.

File: Network/Network.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+#endif
+
+#include <Arduino.h>
+#include <stdio.h>      /* printf, scanf, NULL */
+#include <stdlib.h>     /* malloc, free, rand */
+
+class Network{
+private:
+    String ssid = STASSID;
+    String password = STAPASS;
```
Comment (kuba8686) [inline Network/Network.h:21]: as above (line 4-6)

File: Network/Network.h (hunk 1 above)
Comment (Siwy160) [reply Network/Network.h:6]: Or you can create separate ticket for that and do it later.

File: Network/Network.h [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        Convert from const char* to uint8_t*
+
+        @param char array looks like this: "xx:xx:xx:xx:xx:xx"
+        @return Array of 6 uint8_t elements
+    */
+    uint8_t* str2mac(const char *mac){
+        int bytes[6];
+        uint8_t* mac_bytes = (uint8_t*) malloc(6);
+
+        if( 6 == sscanf( mac, "%x:%x:%x:%x:%x:%x%*c",
```
Comment (cichy1024) [inline Network/Network.h:35]: Why did you remove these `hh` from `%x` and changed spacing?
The removed file was much closer to our coding rules.

Comment (Deelay1403) [reply Network/Network.h:35]: Because "%hhx[...]" didn't work so well. In this format, the mac address had been cut off.
